src/components/Modeler.vue

```diff
@@ -586,6 +588,35 @@ export default {
         }
       });
     },
+
+    getBoundaryEvents(process) {
+      return process.flowElements
+        ? process.flowElements.filter(({$type}) => $type === 'bpmn:BoundaryEvent')
+        : [];
+    },
+    createBoundaryEvent(definition) {
+      let boundaryEvent = this.moddle.create('bpmn:BoundaryEvent', {
+        ...definition,
+        cancelActivity: definition.get('cancelActivity'),
+        attachedToRef: definition.get('attachedToRef'),
+      });
+      boundaryEvent.$parent = definition.$parent;
+      if (definition.get('outgoing').length) {
+        boundaryEvent.set('outgoing', definition.get('outgoing'));
+      }
+      return boundaryEvent;
+    },
+    replaceDefinition(definition, boundaryEvent, process) {
+      const definitionIndex = process.flowElements.indexOf(definition);
+      this.definitions.get('rootElements').find(currentProcess => currentProcess === process).flowElements[definitionIndex] = boundaryEvent;
```

Comment:
Is there a reason mutating `process` directly doesn't work? It seems the following should work just the same:
```js
process.flowElements[definitionIndex] = boundaryEvent;
```

Reply:
Seems to work